### PR TITLE
Tms/build rake task #24

### LINKED ISSUE #24: Rake task for csv files

### PR DESCRIPTION
**Pull request by tmikeschu**
Worked with @Dpalazzari - ready for review

- Adds db associations for each table
- Adds rake tasks for seeding database
- Adds csv data
 * To load, run `rake load_data:(thing)` in the order specified in the rakefile
- Fixed bugs for created/updated_at endpoints

Needs in-depth review, all tests passing

@sallymacnicholas 

Closes #24

### CODE REVIEW

File: Rakefile [hunk 1]
```diff
@@ -2,5 +2,66 @@
...
+  end
+
+  # Refactor if possible?
+  # def import(model)
+  #   data        = File.read("data/#{model + 's'}.csv")
+  #   parsed_data = CSV.parse(data, headers: true)
+  #   parsed_data.each do |row|
+  #     model.camelize.constantize.create(row.to_hash)
+  #   end
+  # end
```
Comment (Dpalazzari) [inline Rakefile:63]: Someday, @tmikeschu. someday.

File: app/models/customer.rb [hunk 2]
```diff
@@ -1,2 +1,5 @@
 class Customer < ApplicationRecord
+  has_many :invoices
+  has_many :merchants, through: :invoices
+  has_many :transactions, through: :invoices
```
Comment (Dpalazzari) [inline app/models/customer.rb:4]: 👍

File: app/models/invoice.rb [hunk 3]
```diff
@@ -1,2 +1,8 @@
 class Invoice < ApplicationRecord
+  has_many :invoice_items
+  has_many :items, through: :invoice_items
+  has_many :transactions
+
+  belongs_to :merchant
+  belongs_to :customer
```
Comment (Dpalazzari) [inline app/models/invoice.rb:7]: 👍

File: app/models/merchant.rb [hunk 4]
```diff
@@ -1,2 +1,7 @@
 class Merchant < ApplicationRecord
+  has_many :items
+  has_many :invoices
+  has_many :invoice_items, through: :invoices
+  has_many :transactions, through: :invoices
+  has_many :customers, through: :invoices
```
Comment (Dpalazzari) [inline app/models/merchant.rb:6]: 👍

File: spec/requests/api/v1/customers/request_spec.rb [hunk 5]
```diff
@@ -145,7 +145,7 @@
   end
 
   it "finds all customers matching created_at timestamps" do
-    create_list(:customer, 3)
+    create_list(:customer, 3, created_at: Time.now)
```
Comment (Dpalazzari) [inline spec/requests/api/v1/customers/request_spec.rb:148]: We should ask @sallymacnicholas about these strange errors/bugs we were getting.

**Comment by Dpalazzari** [comment]
I went thru this with Mike, but I looked over the files again real quick. It all looks good to me!

FYI @sallymacnicholas

File: spec/requests/api/v1/customers/request_spec.rb (hunk 5 above)
Comment (tmikeschu) [reply spec/requests/api/v1/customers/request_spec.rb:148]: @sallymacnicholas : before we were just generating three customers, and most of the time the tests would pass. Every 1 in 5 or so, the times would be different, and only return one or two. Adding the fixed Time.now works, but is there a better way?

**Pull request merged**